spec/features/user_add_contact_spec.rb

```diff
@@ -2,16 +2,34 @@
 
 RSpec.describe "Users", type: :feature do
   before(:each) do
-    @user1 = FactoryBot.create :user
-    @user2 = FactoryBot.create :user
-    @user3 = FactoryBot.create :user
-    sign_in @user1
+    @requesting_user = FactoryBot.create :user
+    @requested_user = FactoryBot.create :user
+  end
+
+  it "adds a contact request when the button is clicked" do
+    sign_in @requesting_user
     visit home_index_path
+    find("form[action='#{user_contact_requests_path(@requested_user)}']").find("input").click
+    expect(@requested_user.contact_requests).to_not be_empty
+    expect(@requested_user.contact_requests).to include(@requesting_user)
   end
 
-  it "adds a contact when the button is clicked" do
-    find_by_id(@user2.id.to_s).click_button
+  it 'is possible to accept a contact request' do
+    sign_in @requested_user
+    @requested_user.contact_requests << @requesting_user
+    visit user_contact_requests_path(@requested_user)
+    click_button("✓")
+    expect(page).to_not have_text(@requesting_user.email)
+    expect(@requesting_user.contacts).to include(@requested_user)
+  end
 
-    expect(@user1.contacts).to include(@user2)
+  it 'is possible to deny a contact request' do
+    sign_in @requested_user
+    @requested_user.contact_requests << @requesting_user
+    visit user_contact_requests_path(@requested_user)
+    find("form[action='#{user_contact_request_path(@requested_user, @requesting_user)}']").find("input").click
+    expect(page).to_not have_text(@requesting_user.email)
+    expect(@requesting_user.contacts).to_not include(@requested_user)
+    expect(@requested_user.contact_requests).to include(@requesting_user)
```

Comment:
Warum bleibt der requesting user in den Contact requests wenn er doch abgelehnt wurde und die Email nicht mehr angezeigt wird?

Reply:
du hast Recht sollte er nicht und war eigentlich nur zum ausprobieren! Das muss ich nochmal fixen und schauen ob der Test noch läuft danke für den Hinweis!

Reply:
ist gefixt